exporter/metrics.go

```diff
@@ -2,30 +2,114 @@ package exporter
+		{"collstats_storage.wiredTiger", "collstats_storage_wt"},
+		{"collstats_storage.indexDetails", "collstats_storage_idx"},
+		{"collStats.storageStats", "collstats_storage"},
+		{"collStats.latencyStats", "collstats_latency"},
+	}
+
+	// This map is used to add labels to some specific metrics.
+	// For example, the fields under the serverStatus.opcounters. structure have this
+	// signature:
+	//
+	//    "opcounters": primitive.M{
+	//        "insert":  int32(4),
+	//        "query":   int32(2118),
+	//        "update":  int32(14),
+	//        "delete":  int32(22),
+	//        "getmore": int32(9141),
+	//        "command": int32(67923),
+	//    },
+	//
+	// Applying the renaming rules, serverStatus will become ss but instead of having metrics
+	// with the form ss.opcounters.<operation> where operation is each one of the fields inside
+	// the structure (insert, query, update, etc), those keys will become labels for the same
+	// metric name. The label name is defined as the value for each metric name in the map and
+	// the value the labeil will have is the field name in the structure. Example.
+	//
+	//   mongodb_ss_opcounters{legacy_op_type="insert"} 4
+	//   mongodb_ss_opcounters{legacy_op_type="query"} 2118
+	//   mongodb_ss_opcounters{legacy_op_type="update"} 14
+	//   mongodb_ss_opcounters{legacy_op_type="delete"} 22
+	//   mongodb_ss_opcounters{legacy_op_type="getmore"} 9141
+	//   mongodb_ss_opcounters{legacy_op_type="command"} 67923
+	//
+	nodeToPDMetrics = map[string]string{ //nolint
+		"collStats.storageStats.indexDetails.":            "index_name",
+		"globalLock.activeQueue.":                         "count_type",
+		"globalLock.locks.":                               "lock_type",
+		"serverStatus.asserts.":                           "assert_type",
+		"serverStatus.connections.":                       "conn_type",
+		"serverStatus.globalLock.currentQueue.":           "count_type",
+		"serverStatus.metrics.commands.":                  "cmd_name",
+		"serverStatus.metrics.cursor.open.":               "csr_type",
+		"serverStatus.metrics.document.":                  "doc_op_type",
+		"serverStatus.opLatencies.":                       "op_type",
+		"serverStatus.opReadConcernCounters.":             "concern_type",
+		"serverStatus.opcounters.":                        "legacy_op_type",
+		"serverStatus.opcountersRepl.":                    "legacy_op_type",
+		"serverStatus.transactions.commitTypes.":          "commit_type",
+		"serverStatus.wiredTiger.concurrentTransactions.": "txn_rw_type",
+		"serverStatus.wiredTiger.perf.":                   "perf_bucket",
+		"systemMetrics.disks.":                            "device_name",
+	}
+
+	// Regular expressions used to make the metric name Prometheus-compatible
+	// This variables are global to compile the regexps only once.
+	specialCharsRe      = regexp.MustCompile(`[^a-zA-Z0-9_]+`) //nolint
+	repeatedUnderscores = regexp.MustCompile(`__+`)            //nolint
```

Comment:
```suggestion
	repeatedUnderscoresRe = regexp.MustCompile(`__+`)            //nolint
```

Non consistent naming.